store/postgres/src/deployment_store.rs

```diff
@@ -1118,6 +1118,16 @@ impl DeploymentStore {
                     )?;
                 }
 
+                if !batch.deterministic_errors.is_empty() && batch.is_non_fatal_errors_active {
```

Comment:
It would be better to pull this `if` into the previous `if` statement so that you only have to say `if batch.is_non_fatal_errors_active { .. }`